app/src/main/java/ar/scacchipa/jsonplaceholderviewer/data/RetrofitRepository.kt

```diff
@@ -0,0 +1,23 @@
+package ar.scacchipa.jsonplaceholderviewer.data
+
+class CommentRepository(
+     var apiSourceData: ICommentSourceData// = provideNetworkApi(provideRetrofit())
+): ICommentRepository {
+    override suspend fun getPosts(): List<Post> {
+        val response = apiSourceData.getPosts()
+        return if (response.isSuccessful) {
+            response.body() ?: listOf()
+        } else {
+            listOf()
+        }
+    }
+    override suspend fun getComment(id: Int): List<Comment> {
+        apiSourceData.getComment(id.toString()).let { response ->
+            return if (response.isSuccessful) {
```

Comment:
There isn't context in this scope. Maybe it would be better to throw a exception.

Reply:
mm i dont think you need a context , ideally you should be able to emit from the vm to the views objects that represent ui state, even if that ui state means show an error dialog, a toast or a retry button. The context for the toast is a ui problem, so it is activity/fragment/commposable responsability. Ideally from domain/data layers should be emmited objects of kind Result<T> for example, which If i dont remember wrong are part of kotlin now. Result<T> is like a box, a wrapper, there can go a success case (a comment pojo object for example) or a failure case ( a throwable). Other approach would be using mvi, and observe in the activity ui state objects that encapsule all the info you. need, so instead of having a vm emiting list of post, you would have a vm emiting ui state objects, which contains a list of post and other info relevant to display for example, a boolean for an error, or a boolean for loading state. But we can go deeper about this different approaches on friday, the main idea I want to transfer for this comment is that you dont need the context here in this class for achiving a toast on the ui.